src/games/strategy/engine/framework/GameRunner.java

```diff
@@ -177,37 +190,28 @@ private static void showMainFrame() {
     });
   }
 
+
+
   /**
    * Move command line arguments to System.properties
    */
-  private static void handleCommandLineArgs(final String[] args) {
-    final String[] properties = getProperties();
-    // if only 1 arg, it might be the game path, find it (like if we are double clicking a savegame)
-    // optionally, it may not start with the property name
-    if (args.length == 1) {
-      boolean startsWithPropertyKey = false;
-      for (final String prop : properties) {
-        if (args[0].startsWith(prop)) {
-          startsWithPropertyKey = true;
-          break;
-        }
-      }
-      if (!startsWithPropertyKey) {
-        // change it to start with the key
-        args[0] = TRIPLEA_GAME_PROPERTY + "=" + args[0];
-      }
+  public static void handleCommandLineArgs(final String[] args, final String[] availableProperties, GameMode gameMode) {
+    if (args.length == 1 && !argsStartsWithPropertykey(args[0])) {
```

Comment:
The value of `argsStartsWithPropertykey` was extracted to a  method, simplified the block compared to to the roughly 7 lines it took before.